captum/attr/_utils/common.py

```diff
@@ -422,28 +621,36 @@ def _expand_target(target, n_steps, expansion_type=ExpansionTypes.repeat):
 
 
 def _call_custom_attribution_func(
-    custom_attribution_func, multipliers, inputs, baselines
-):
+    custom_attribution_func: Callable[..., Tuple[Tensor, ...]],
+    multipliers: Tuple[Tensor, ...],
+    inputs: Tuple[Tensor, ...],
+    baselines: Tuple[Tensor, ...],
+) -> Tuple[Tensor, ...]:
     assert callable(custom_attribution_func), (
         "`custom_attribution_func`"
         " must be a callable function but {} provided".format(
             type(custom_attribution_func)
         )
     )
     custom_attr_func_params = signature(custom_attribution_func).parameters
-    assert len(custom_attr_func_params) in range(1, 4), (
-        "`custom_attribution_func`" " must take at least one and at most 3 arguments"
-    )
 
     if len(custom_attr_func_params) == 1:
         return custom_attribution_func(multipliers)
     elif len(custom_attr_func_params) == 2:
         return custom_attribution_func(multipliers, inputs)
     elif len(custom_attr_func_params) == 3:
         return custom_attribution_func(multipliers, inputs, baselines)
+    else:
+        raise AssertionError(
+            "`custom_attribution_func` must take at least one and at most 3 arguments."
+        )
 
 
-def _extract_device(module, hook_inputs, hook_outputs):
+def _extract_device(
+    module: Module,
+    hook_inputs: Union[Tensor, Tuple[Tensor, ...]],
```

Comment:
`hook_inputs` can be None too, right ?

Here too, maybe: hook_outputs:Union[None, Tensor, Tuple[Tensor, ...]] is clearner